boards/stm32f4discovery/include/periph_conf.h

```diff
@@ -115,39 +117,16 @@ extern "C" {
  * @name ADC configuration
  * @{
  */
-#define ADC_NUMOF           (2U)
-#define ADC_0_EN            1
-#define ADC_1_EN            1
-#define ADC_MAX_CHANNELS    2
-
-/* ADC 0 configuration */
-#define ADC_0_DEV           ADC1
-#define ADC_0_CHANNELS      2
-#define ADC_0_CLKEN()       (RCC->APB2ENR |= RCC_APB2ENR_ADC1EN)
-#define ADC_0_CLKDIS()      (RCC->APB2ENR &= ~(RCC_APB2ENR_ADC1EN))
-#define ADC_0_PORT          GPIOA
-#define ADC_0_PORT_CLKEN()  (RCC->AHB1ENR |= RCC_AHB1ENR_GPIOAEN)
-/* ADC 0 channel 0 pin config */
-#define ADC_0_CH0           1
-#define ADC_0_CH0_PIN       1
-/* ADC 0 channel 1 pin config */
-#define ADC_0_CH1           4
-#define ADC_0_CH1_PIN       4
-
-/* ADC 1 configuration */
-#define ADC_1_DEV           ADC2
-#define ADC_1_CHANNELS      2
-#define ADC_1_CLKEN()       (RCC->APB2ENR |= RCC_APB2ENR_ADC2EN)
-#define ADC_1_CLKDIS()      (RCC->APB2ENR &= ~(RCC_APB2ENR_ADC2EN))
+static const adc_conf_t adc_config[] = {
+    /* device, pin, channel, RCC bit */
+    {ADC1, GPIO_PIN(PORT_A, 1), 1, 8},
+    {ADC1, GPIO_PIN(PORT_A, 4), 4, 8},
+    {ADC2, GPIO_PIN(PORT_C, 1), 11, 9},
+    {ADC2, GPIO_PIN(PORT_C, 2), 12, 9}
+};
```

Comment:
And adding a line like below?

    #define ADC_LINE(adc, gpio, adc_channel)           { adc, gpio,
adc_channel, adc == ADC1 ? 8 : adc == ADC2 ? 9 : 15 }

?

On Tue, Dec 8, 2015 at 4:13 PM, Hauke Petersen <notifications@github.com>
wrote:

> In boards/stm32f4discovery/include/periph_conf.h
> <https://github.com/RIOT-OS/RIOT/pull/4430#discussion_r46964553>:
>
> > -/* ADC 0 channel 1 pin config */
> > -#define ADC_0_CH1           4
> > -#define ADC_0_CH1_PIN       4
> > -
> > -/* ADC 1 configuration */
> > -#define ADC_1_DEV           ADC2
> > -#define ADC_1_CHANNELS      2
> > -#define ADC_1_CLKEN()       (RCC->APB2ENR |= RCC_APB2ENR_ADC2EN)
> > -#define ADC_1_CLKDIS()      (RCC->APB2ENR &= ~(RCC_APB2ENR_ADC2EN))
> > +static const adc_conf_t adc_config[] = {
> > +    /* device, pin, channel, RCC bit */
> > +    {ADC1, GPIO_PIN(PORT_A, 1), 1, 8},
> > +    {ADC1, GPIO_PIN(PORT_A, 4), 4, 8},
> > +    {ADC2, GPIO_PIN(PORT_C, 1), 11, 9},
> > +    {ADC2, GPIO_PIN(PORT_C, 2), 12, 9}
> > +};
>
> Actually that is what I started with yesterday night, but I came to the
> conclusion that the slight duplication of some ADCx devices and RCC bits is
> well spend for getting a much less complex way for defining the lines. In
> terms of code size we get also a simpler implementation, so the reduced
> code there was more than we need to store those duplicate ADC defs and RCC
> pins...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/4430/files#r46964553>.
>